lib/types/inet-address.js

```diff
@@ -10,238 +11,243 @@ const utils = require("../utils");
+        for (let i = 0; i < this.buffer.length; i = i + 2) {
+            if (this.buffer[i] === 0 && this.buffer[i + 1] === 0) {
+                //its a group of zeros
+                if (start < 0) {
+                    start = i;
+                }
+
+                // at the end of the buffer, make a final call to checkLongest.
+                if (i === this.buffer.length - 2) {
+                    checkLongest(i + 2);
+                }
+                continue;
             }
-            continue;
+            //its a group of non-zeros
+            checkLongest(i);
         }
-        //its a group of non-zeros
-        checkLongest(i);
-    }
 
-    let address = "";
-    for (let h = 0; h < this.buffer.length; h = h + 2) {
-        if (h === longest.start) {
-            address += ":";
-            continue;
-        }
-        if (h < longest.start + longest.length && h > longest.start) {
-            //its a group of zeros
-            continue;
+        let address = "";
+        for (let h = 0; h < this.buffer.length; h = h + 2) {
+            if (h === longest.start) {
+                address += ":";
+                continue;
+            }
+            if (h < longest.start + longest.length && h > longest.start) {
+                //its a group of zeros
+                continue;
+            }
+            if (address.length > 0) {
+                address += ":";
+            }
+            address += ((this.buffer[h] << 8) | this.buffer[h + 1]).toString(
+                16,
+            );
         }
-        if (address.length > 0) {
+        if (address.charAt(address.length - 1) === ":") {
             address += ":";
         }
-        address += ((this.buffer[h] << 8) | this.buffer[h + 1]).toString(16);
-    }
-    if (address.charAt(address.length - 1) === ":") {
-        address += ":";
+        return address;
     }
-    return address;
-};
 
-/**
- * Returns the string representation.
- * Method used by the native JSON.stringify() to serialize this instance.
- */
-InetAddress.prototype.toJSON = function () {
-    return this.toString();
-};
+    /**
+     * Returns the string representation.
+     * Method used by the native JSON.stringify() to serialize this instance.
+     */
+    toJSON() {
+        return this.toString();
+    }
 
-/**
- * Validates for a IPv4-Mapped IPv6 according to https://tools.ietf.org/html/rfc4291#section-2.5.5
- * @private
- * @param {Buffer} buffer
- */
-function isValidIPv4Mapped(buffer) {
-    // check the form
-    // |      80 bits   | 16 |   32 bits
-    // +----------------+----+-------------
-    // |0000........0000|FFFF| IPv4 address
-
-    for (let i = 0; i < buffer.length - 6; i++) {
-        if (buffer[i] !== 0) {
-            return false;
+    /**
+     * Validates for a IPv4-Mapped IPv6 according to https://tools.ietf.org/html/rfc4291#section-2.5.5
+     * @private
+     * @param {Buffer} buffer
+     */
+    static isValidIPv4Mapped(buffer) {
```

Comment:
This function wasn't exposed in the old api. Is this a mistake or deliberate change to add it to inet class?

Reply:
I didn't saw that function when I was rewriting DataStax code into JS class. I will remove that as it is never used and there isn't any mention about this in docs.